connectors/email/src/main/java/io/camunda/connector/email/client/jakarta/inbound/PollingManager.java

```diff
@@ -150,62 +214,62 @@ private void processMail(IMAPMessage message, PollingConfig pollingConfig) {
     // Setting `peek` to true prevents the library to trigger any side effects when reading the
     // message, such as marking it as read
     message.setPeek(true);
-    this.correlateEmail(message, connectorContext);
+    Email email = this.jakartaUtils.createEmail(message);
+    boolean markAsProcessed = process(email);
     message.setPeek(false);
-    switch (pollingConfig.handlingStrategy()) {
-      case READ -> this.jakartaUtils.markAsSeen(message);
-      case DELETE -> this.jakartaUtils.markAsDeleted(message);
-      case MOVE -> {
-        this.jakartaUtils.markAsSeen(message);
-        this.jakartaUtils.moveMessage(this.store, message, pollingConfig.targetFolder());
+    if (markAsProcessed) {
+      switch (pollingConfig.handlingStrategy()) {
+        case READ -> this.jakartaUtils.markAsSeen(message);
+        case DELETE -> this.jakartaUtils.markAsDeleted(message);
+        case MOVE -> {
+          this.jakartaUtils.markAsSeen(message);
+          this.jakartaUtils.moveMessage(this.store, message, pollingConfig.targetFolder());
+        }
       }
     }
   }
 
-  private void correlateEmail(Message message, InboundConnectorContext connectorContext) {
-    Email email = this.jakartaUtils.createEmail(message);
-    List<Document> documents = this.createDocumentList(email, connectorContext);
-    connectorContext.correlate(
-        CorrelationRequest.builder()
-            .variables(
-                new ReadEmailResponse(
-                    email.messageId(),
-                    email.from(),
-                    email.headers(),
-                    email.subject(),
-                    email.size(),
-                    email.body().bodyAsPlainText(),
-                    email.body().bodyAsHtml(),
-                    documents,
-                    email.receivedAt()))
-            .messageId(email.messageId())
-            .build());
-  }
-
-  private List<Document> createDocumentList(Email email, InboundConnectorContext connectorContext) {
-    if (!(connectorContext.canActivate(
-            new ReadEmailResponse(
-                email.messageId(),
-                email.from(),
-                email.headers(),
-                email.subject(),
-                email.size(),
-                email.body().bodyAsPlainText(),
-                email.body().bodyAsHtml(),
-                List.of(),
-                email.receivedAt()))
-        instanceof ActivationCheckResult.Success)) {
-      return List.of();
-    } else
-      return email.body().attachments().stream()
-          .map(
-              document ->
-                  connectorContext.create(
-                      DocumentCreationRequest.from(document.inputStream())
-                          .contentType(document.contentType())
-                          .fileName(document.name())
-                          .build()))
-          .toList();
+  private boolean process(Email email) {
```

Comment:
I think this naming is confusing. This function is named `process` and has an Email as input parameter and above we have the function processEmail. Also it is unclear / confusing, what these booleans are that these mails return.

I think it would be clearer what is happening here, if we have functions that do the evaluation, like canActivate, canCorrelate. And the functions that do the actual workload.